File: 04web_iot/02dht11.py
from flask import Flask, request, redirect
import RPi.GPIO as GPIO
import Adafruit_DHT, json, time

app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret'

GPIO.setmode(GPIO.BCM)
sensor = Adafruit_DHT.DHT11
pin_dht11 = 24

@app.route('/')
def main():
    return redirect('/static/dht11.html')

@app.route('/monitor')
def monitoring():
    try:
        print '/monitor'
        humidity, temperature = Adafruit_DHT.read_retry(sensor, pin_dht11)
        obj = {'humi' : humidity, 'temp' : temperature}
        return json.dumps(obj)
    except Exception as e:
        print 'err', e


if __name__ == '__main__':
    try:
        app.run(host='0.0.0.0')
    finally:
        print 'cleaning up'
        GPIO.cleanup()

File: 04web_iot/02dht11.js
var express = require('express');
var app = express();
var sensor = require('node-dht-sensor');
var dht_type = 11;
var pin_dht = 24

app.use(express.static(__dirname + '/static'));
app.get('/', function(req, res){
	res.redirect('/dht11.html');
});

app.get('/monitor', function(req, res){
	console.log(req.url, req.query);
	sensor.read(dht_type, pin_dht, function(err, temperature, humidity){
		if(!err){
			var obj = {'humi' : humidity, 'temp' : temperature};
		}else{
			var obj = {'humi' : 'err', 'temp' : 'err'};
		}
		console.log(JSON.stringify(obj));
		res.send(obj);
	});
});


app.listen(5000, function(){
	console.log('server running on 5000.');
});